File: JavaScript/Algorithms/insertionSort.js
const insertionSort = (array) => {
    for(let i=1;i<array.length;i++) {
        let key = array[i];
        let j = i - 1;

        while(j>=0 && array[j]>key) {
            array[j+1] = array[j];
            j--;
        }
        array[j+1] = key;
    }
    return array;
}

let array = [5, 4, 3, 2, 1];

console.log(insertionSort(array));

File: Julia/Algorithms/insertion_sort.jl
function insertion_sort(array)
    for i in 1:length(array)
        key = array[i]
        j = i - 1

        while j >= 1 && array[j] > key
            array[j+1] = array[j]
            j -= 1
        end

        array[j+1] = key
    end
    return array
end

array = [-5, 4, 3, 2, 10]

println(insertion_sort(array))